tests/data/instances/text_classification/frame_instance_test.py

```diff
@@ -0,0 +1,86 @@
+# pylint: disable=no-self-use,invalid-name
+from deep_qa.data import DataIndexer
+
+from deep_qa.data.instances.text_classification.frame_instance import FrameInstance, IndexedFrameInstance
+from tests.common.test_case import DeepQaTestCase
+
+
+class TestFrameInstance(DeepQaTestCase):
+
+    def setUp(self):
+        super(TestFrameInstance, self).setUp()
+        # Example of a typical input
+        self.line = "event:plant absorb water###" \
+                    "participant:water###" \
+                    "agent:plant###" \
+                    "finalloc:soil"\
+                    + "\t" + "finalloc:soil"
+        self.line_with_no_label_val = "event:plant absorb water###" \
+                    "participant:water###" \
+                    "agent:plant###" \
+                    "finalloc:soil" \
+                    + "\t" + "finalloc"
+        self.padded_slots = ['plant', 'unk', 'unk', 'unk', 'unk',
+                             'plant absorb water', 'ques', 'unk', 'unk',
+                             'unk', 'unk', 'unk', 'unk', 'unk', 'unk',
+                             'unk', 'unk', 'unk', 'unk', 'unk', 'unk', 'unk',
+                             'unk', 'unk', 'unk', 'unk', 'water']
+        self.data_indexer = DataIndexer()
+        for word in ['plant', 'unk', 'absorb', 'ques', 'water', 'soil']:
+            self.data_indexer.add_word_to_index(word)
+
+    def test_convert_instance_to_indexed_instance(self):
+        instance = FrameInstance.read_from_line(self.line)
+        indexed_instance = instance.to_indexed_instance(self.data_indexer)
+        assert indexed_instance.label == [self.data_indexer.get_word_index('soil')]
+
+    def test_convert_instance_no_label_value_to_indexed_instance(self):
+        instance = FrameInstance.read_from_line(self.line_with_no_label_val)
+        indexed_instance = instance.to_indexed_instance(self.data_indexer)
+        assert indexed_instance.label == [self.data_indexer.get_word_index('soil')]
+
+    def test_slots_unwrap_correctly(self):
+        instance = FrameInstance.read_from_line(self.line)
+        # what we construct
+        machine_label = instance.label
+        machine_slot_values = instance.text
+        # what we expect
+        expected_label = "soil"
+        # do they match?
+        assert machine_label == expected_label
+        assert machine_slot_values == self.padded_slots
```

Comment:
The label here is what you're trying to predict.  You want to have the model output "soil".  But it looks like there's nothing in this instance that tells the model what slot you're trying to query.  How would a model know which slot to fill, when it's trying to predict "soil"?